flash/image/classification/integrations/labelstudio/handler.py

```diff
@@ -0,0 +1,148 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+import os
+from typing import Any, Dict, Optional
+
+import numpy as np
+import requests
+import torch
+
+import flash
+from flash.core.classification import Probabilities
+from flash.core.utilities.imports import _BAAL_AVAILABLE, _LABEL_STUDIO_ML_AVAILABLE, requires
+from flash.image import ImageClassificationData, ImageClassifier
+from flash.image.classification.integrations.baal.dropout import InferenceMCDropoutTask
+
+if _BAAL_AVAILABLE:
+    from baal.active.heuristics import BALD
+
+if _LABEL_STUDIO_ML_AVAILABLE:
+    from label_studio_ml.model import LabelStudioMLBase
+    from label_studio_ml.utils import get_choice, get_single_tag_keys, is_skipped
+
+else:
+
+    class LabelStudioMLBase:
+        pass
+
+
+class ImageClassifierAPI(LabelStudioMLBase):
+
+    """This class is deploy a Flash Classifier to be used as ML backend to LabelStudio."""
+
+    _IMAGE_CACHE_DIR: str
+
+    requires("baal")
+    requires("label_studio_ml")
+
+    def __init__(self, source: str, **kwargs):
+        super().__init__(**kwargs)
+        self.from_name, self.to_name, self.value, self.classes = get_single_tag_keys(
+            self.parsed_label_config, "Choices", "Image"
+        )
+
+        num_classes = len(self.classes)
+
+        head = torch.nn.Sequential(
+            torch.nn.Dropout(p=0.1),
+            torch.nn.Linear(512, num_classes),
+        )
+
+        self.model = ImageClassifier(
+            backbone="resnet18", num_classes=num_classes, head=head, serializer=Probabilities()
+        )
+
+        self.predict_model = InferenceMCDropoutTask(self.model, 2)
+
+        self.heuristic = BALD()
+
+        self._IMAGE_CACHE_DIR = os.path.join(os.path.dirname(source), "image_cache")
+        os.makedirs(self._IMAGE_CACHE_DIR, exist_ok=True)
+
+    def predict(self, tasks, **kwargs):
+        image_urls = [self._firstv(task["data"]) for task in tasks]
+        predict_files = [self._download_image(url) for url in image_urls]
+
+        datamodule = ImageClassificationData.from_files(predict_files=predict_files, batch_size=2)
+        trainer = flash.Trainer(max_epochs=1, gpus=int(torch.cuda.is_available()))
```

Comment:
This is pretty hardcoded, any way this can be exposed?

Reply:
Not yet.

Reply:
But there might be some solutions.